test/e2e/lib/helpers/broker_event_transformation_test_with_source.go

```diff
@@ -165,14 +166,43 @@ func BrokerEventTransformationMetricsTestHelper(client *lib.Client, projectID st
 				lib.E2EDummyRespEventType: 1,
 			},
 		},
-		lib.TriggerMetricAssertion{
-			ProjectID:       projectID,
-			BrokerName:      brokerName,
-			BrokerNamespace: client.Namespace,
-			StartTime:       start,
-			CountPerTrigger: map[string]int64{
-				trigger.Name:     1,
-				respTrigger.Name: 1,
+		lib.TriggerMetricWithRespCodeAssertion{
```

Comment:
The downside to structuring the assertion in this way is that it will not be able to detect if metrics are emitted with some other response code whereas previously this would have failed the test. An alternative approach would be to add the response code to the count map key.

Reply:
Good point.